### ISSUE TITLE
Add support for new experiments API changes

### ISSUE THREAD

**Issue by jyu00** [Qiskit/qiskit-ibmq-provider#854]
### What is the expected enhancement?

Support the new experiments API changes:

- ~[ ] make `device_components` optional for analysis results~
- ~[ ] make `fit` optional for analysis results~
- [x] support updating experiment `extra` and `tags`
- [x] support device backend name
- [x] support excluder_owner and owner_only when filtering experiments
- [x] typo in `public_only` docstring
- [x] new `verified` boolean
- [x] new free-form `notes` in experiments

**Comment by mriedem** [comment]
It's now also possible to filter experiments in `GET /experiments` by `owner` using the special `me` and `!me` values:

- `GET /experiments?owner=me`: only return experiments owned by the user making the request
- `GET /experiments?owner=!me`: only return experiments available to but not owned by the user making the request

**Comment by jyu00** [comment]
@mriedem FYI since these new changes are to support Ignis, I'm going to wait until the Ignis interface (Qiskit/qiskit-terra#5499) is finalized before working on this, since there will be a bit of interface change.

**Comment by mriedem** [comment]
OK, no problem. Would you want to split out the `owner=me` and `owner=!me` support stuff into a separate issue to be worked on outside of the Ignis changes, or is just everything held up on that interface change?

**Comment by jyu00** [comment]
Sure. `owner=me` can be done separately.

**Comment by jyu00** [comment]
Closed by #945
